programs/fundamentals/cp.scl

```diff
@@ -0,0 +1,116 @@
+
+
+
+// U-00000000 … U-0000007F | 0xxxxxxx 
+// U-00000080 … U-000007FF | 110xxxxx 10xxxxxx 
+// U-00000800 … U-0000FFFF | 1110xxxx 10xxxxxx 10xxxxxx 
+// U-00010000 … U-0010FFFF | 11110xxx 10xxxxxx 10xxxxxx 10xxxxxx 
+
+ord(char,outer(__map))->(
+    blist=[];
+    for(split(encode_b64(char)),
+        put(blist, null, __map:_, 'extend')
+    );
+    //print(blist);
+    firstZero=first(blist,if(!_,break(_i)));
+    digs=if(length(blist)==2*6 &&firstZero==0,[1,2,3,4,5,6,7],
+    length(blist)==3*6&&firstZero==2,[3,4,5,6,7,10,11,12,13,14,15],
+    length(blist)==4*6&&firstZero==3,[4,5,6,7,10,11,12,13,14,15,18,19,20,21,22,23],
+    length(blist)==6*6&&firstZero==4,[5,6,7,10,11,12,13,14,15,18,19,20,21,22,23,26,27,28,29,30,31],
+    return('ERROR!!')
+    );
+    res=0;
+    for(digs,
+    res=res*2;
+    res=res+blist:_
+    );
+    res
+);
+
+ord_string(str)->(
+    if (length(str)<=1,return(map(split(str),ord(_))));
+
+    c_for(i=1,i<=length(str),i+=1,
+
+    o=ord(slice(str,0,i));
+
+    if (chr(o)==slice(str,0,i),
+
+        res=[o];
+        if(i<length(str),put(res, null, ord_string(slice(str,i)), 'extend'));
+        return(res)
+
+
+    );
+
+
+    )
+
+
+);
+
+
+//chr
+__rmap={[0, 0, 0, 0, 0, 0]-> 'A', [0, 0, 0, 0, 0, 1]-> 'B', [0, 0, 0, 0, 1, 0]-> 'C', [0, 0, 0, 0, 1, 1]-> 'D', [0, 0, 0, 1, 0, 0]-> 'E', [0, 0, 0, 1, 0, 1]-> 'F', [0, 0, 0, 1, 1, 0]-> 'G', [0, 0, 0, 1, 1, 1]-> 'H', [0, 0, 1, 0, 0, 0]-> 'I', [0, 0, 1, 0, 0, 1]-> 'J', [0, 0, 1, 0, 1, 0]-> 'K', [0, 0, 1, 0, 1, 1]-> 'L', [0, 0, 1, 1, 0, 0]-> 'M', [0, 0, 1, 1, 0, 1]-> 'N', [0, 0, 1, 1, 1, 0]-> 'O', [0, 0, 1, 1, 1, 1]-> 'P', [0, 1, 0, 0, 0, 0]-> 'Q', [0, 1, 0, 0, 0, 1]-> 'R', [0, 1, 0, 0, 1, 0]-> 'S', [0, 1, 0, 0, 1, 1]-> 'T', [0, 1, 0, 1, 0, 0]-> 'U', [0, 1, 0, 1, 0, 1]-> 'V', [0, 1, 0, 1, 1, 0]-> 'W', [0, 1, 0, 1, 1, 1]-> 'X', [0, 1, 1, 0, 0, 0]-> 'Y', [0, 1, 1, 0, 0, 1]-> 'Z', [0, 1, 1, 0, 1, 0]-> 'a', [0, 1, 1, 0, 1, 1]-> 'b', [0, 1, 1, 1, 0, 0]-> 'c', [0, 1, 1, 1, 0, 1]-> 'd', [0, 1, 1, 1, 1, 0]-> 'e', [0, 1, 1, 1, 1, 1]-> 'f', [1, 0, 0, 0, 0, 0]-> 'g', [1, 0, 0, 0, 0, 1]-> 'h', [1, 0, 0, 0, 1, 0]-> 'i', [1, 0, 0, 0, 1, 1]-> 'j', [1, 0, 0, 1, 0, 0]-> 'k', [1, 0, 0, 1, 0, 1]-> 'l', [1, 0, 0, 1, 1, 0]-> 'm', [1, 0, 0, 1, 1, 1]-> 'n', [1, 0, 1, 0, 0, 0]-> 'o', [1, 0, 1, 0, 0, 1]-> 'p', [1, 0, 1, 0, 1, 0]-> 'q', [1, 0, 1, 0, 1, 1]-> 'r', [1, 0, 1, 1, 0, 0]-> 's', [1, 0, 1, 1, 0, 1]-> 't', [1, 0, 1, 1, 1, 0]-> 'u', [1, 0, 1, 1, 1, 1]-> 'v', [1, 1, 0, 0, 0, 0]-> 'w', [1, 1, 0, 0, 0, 1]-> 'x', [1, 1, 0, 0, 1, 0]-> 'y', [1, 1, 0, 0, 1, 1]-> 'z', [1, 1, 0, 1, 0, 0]-> '0', [1, 1, 0, 1, 0, 1]-> '1', [1, 1, 0, 1, 1, 0]-> '2', [1, 1, 0, 1, 1, 1]-> '3', [1, 1, 1, 0, 0, 0]-> '4', [1, 1, 1, 0, 0, 1]-> '5', [1, 1, 1, 0, 1, 0]-> '6', [1, 1, 1, 0, 1, 1]-> '7', [1, 1, 1, 1, 0, 0]-> '8', [1, 1, 1, 1, 0, 1]-> '9', [1, 1, 1, 1, 1, 0]-> '+', [1, 1, 1, 1, 1, 1]-> '/'};
```

Comment:
Same here for the descriptive name, just so people understand

Reply:
add 2 comments for that